### PR TITLE
NNC Dynamic Channels last fixes

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72077
* #72036
* #72038
* #72040
* __->__ #72032
* #71651
* #71650
* #71173

This contains a few channels last changes from benchmarking:
- dont permute back to channels last on dynamic, cpu, perf is not good, and use cases for it are exotic atm
- remove the conditional one handling in permutting channels last symbolic tensor on cuda, it's not needed in the permutation case as tests show
- removing logic in torch/csrc/jit/tensorexpr/loopnest.cpp preventing inlining. the condition in checks is always valid given valid construction of ir

I can split up as needed.

Differential Revision: [D33864652](https://our.internmc.facebook.com/intern/diff/D33864652)

### CODE REVIEW

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33864652).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33864652).

File: torch/csrc/jit/passes/symbolic_shape_runtime_fusion.cpp [hunk 1]
```diff
@@ -163,7 +163,8 @@ StrideInput summarizeOutputStrides(const TensorType& tt) {
   // We only try to maintain output striding for channels last tensors,
   // otherwise we defer to contiguous
   // TODO: channels last 3d
-  if (c10::is_channels_last_strides_2d(sizes, strides)) {
+  // NNC Channels last permutation for outputs causes slowdown, disable
```
Comment (navahgar) [inline torch/csrc/jit/passes/symbolic_shape_runtime_fusion.cpp:166]: Is this case still slow after the perf improvements we did?

**Review by navahgar — APPROVED** [review]
LGTM

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33864652).

File: torch/csrc/jit/passes/symbolic_shape_runtime_fusion.cpp (hunk 1 above)
Comment (eellison) [reply torch/csrc/jit/passes/symbolic_shape_runtime_fusion.cpp:166]: yea, for cpu it is

**Pull request closed without merging**